src/geom/bounding_box.C

```diff
@@ -62,6 +62,62 @@ bool BoundingBox::contains_point (const Point & p) const
 }
 
 
+
+Real BoundingBox::max_size() const
+{
+  Real size = 0;
+  for (unsigned int d = 0; d != LIBMESH_DIM; ++d)
+    {
+      Real sized = this->second(d) - this->first(d);
+      if (!libmesh_isinf(sized))
+        size = std::max(size, sized);
+    }
+  return size;
+}
+
+
+
+bool BoundingBox::contains_point
+  (const Point & p, Real abs_tol, Real rel_tol) const
+{
+  libmesh_assert_greater_equal(abs_tol, Real(0));
+  libmesh_assert_greater_equal(rel_tol, Real(0));
+
+  // Just use the other contains_point overload
+  libmesh_assert_greater(rel_tol+abs_tol, Real(0));
+
+  // Find absolute tolerance from relative tolerance
+  const Real tol = std::max(abs_tol, this->max_size()*rel_tol);
+
+  // Make local variables first to make things more clear in a moment
+  const Real my_min_x = this->first(0) - tol;
+  const Real my_max_x = this->second(0) + tol;
+  bool x_int = is_between(my_min_x, p(0), my_max_x);
```

Comment:
```suggestion
  const bool x_int = is_between(my_min_x, p(0), my_max_x);
```

Reply:
You could early return if `x_int = false` and so on through this function if you wanted to avoid some work